xds/server.go

```diff
@@ -317,18 +322,66 @@ func (s *GRPCServer) GracefulStop() {
 	}
 }
 
+func routeAndProcess(ctx context.Context) error {
+	conn := transport.GetConnection(ctx)
+	cw, ok := conn.(*server.ConnWrapper)
+	if !ok {
+		return errors.New("missing connection wrapper in incoming context")
+	}
+	mn, ok := grpc.Method(ctx)
+	if !ok {
+		return errors.New("missing method name in incoming context")
+	}
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		return errors.New("missing metadata in incoming context")
+	}
+	// Assuming to just take the first value defined.
+	authority := md.Get(":authority")
```

Comment:
Does the spec define the behavior when there is more than one `authority` specified?